### DISCUSSION TITLE
Sluggish Browser Performance: Cannot autolaunch d-bus (dbus) without X11 $DISPLAY

### DISCUSSION THREAD

**Discussion by zenmindai** [coder/code-server#6089] [Q&A]
I recently got my code-server running on Termux, which is running on Android 13, but I'm having sluggish UI performance. While I'm running on a S22 Ultra, this seems to be consistent across various browsers. Looking through the logs, I found this: 
`Using the in-memory credential store as the operating system's credential store could not be accessed. Please see https://aka.ms/vscode-server-keyring on how to set this up. Details: Cannot autolaunch D-Bus without X11 $DISPLAY`

From the Termux side, any recommendations for setting up the dbus and X11 for the code-server? I didn't think we would have to setup an X11 display, so this is kinda new for me.

Of course, this is my most recent unresolved troubleshooting issue. If anyone think of something else to look into, please let me know.

**Comment by code-asher** [comment]
That would probably not cause sluggish performance, it just means that any stored secrets will be lost when code-server restarts.

I am not sure why the keyring requires a display; maybe it is to show the password prompt or something.  In the past I have had luck by unlocking the keyring beforehand then launching code-server like this:

```
dbus-run-session sh -c 'echo pass | gnome-keyring-daemon --unlock --replace ; code-server'
```

But unfortunate that the password has to be provided that way.

I also have no idea if this would work on Termux.
